src/pages/profiles/coach/index.tsx

```diff
@@ -27,9 +31,73 @@ const LoginPage = dynamic(() => import('../../sign_in/coach'));
 
 export default function ProfileCoach() {
   const router = useRouter();
+  const [coachDetailProfile, setCoachDetailProfile] =
+    useState<ICoachSubscription | null>(null);
+
+  console.log('====================================');
+  console.log('[ProfileCoach] coachDetailProfile => ', coachDetailProfile);
+  console.log('====================================');
+
+  const [isSubscription, setIsSubscription] = useState<boolean>(false);
+
+  useQuery<ICoachSubscription, ErrorConstructor>(
+    ['coachSubscription'],
+    async () => {
+      const request = coachSubscriptionApi.getSubscription();
+      const result = await request;
+      console.log('[coach subscription] coach result', result);
+      if (result.is_active) {
+        setIsSubscription(true);
+      } else {
+        setIsSubscription(false);
+      }
+      setCoachDetailProfile(result);
+      return result;
+    },
+    {}
+  );
+
+  const [isSubscriptionSuccess, setIsSubscriptionSuccess] =
+    useState<boolean>(false);
+
+  const [isSubscriptionCancel, setIsSubscriptionCancel] =
+    useState<boolean>(false);
```

Comment:
I don't like the idea of having separate states of subscription. Such implementation leaves place to a moment when 
both `isSubscriptionSuccess` and `isSubscriptionCancel` are true, thus leading to unpredictable buggy behaviour. 

I would suggest to create a single state variable which could have only 3 possible values: `PENDING` from state before we check subscription, `SUCCESS` for successful subscription or `CANCEL` for cancelled sub scenario. and build all our business logic around this single source of truth.